lib/solid_queue/processes/registrable.rb

```diff
@@ -53,7 +53,10 @@ def stop_heartbeat
       end
 
       def heartbeat
-        process.heartbeat
+        process.with_lock(&:heartbeat)
```

Comment:
Looks like witch_lock doesnt pass the object to the block. Seeing a `SolidQueue-0.8.2 Error in thread (0.0ms)  error: "ArgumentError no receiver given"` error. 
```
process.with_lock { process.heartbeat }
```

Reply:
D'oh, I'm dumb, seriously. This was working on Sunday and I decided to break it yesterday without testing 😅 I'll fix it.

Reply:
Huh, no, that was correct 😕 I was wondering why the tests were passing... and [this is how `with_lock` is supposed to be used](https://github.com/rails/rails/blob/e09dd95cabbfd1f0357eb4eec79e2ce540ad3d47/activerecord/lib/active_record/locking/pessimistic.rb#L45-L55). The error must come from something else, I'll investigate and fix it.

Reply:
Ah, no, no, your version is the correct one 😆  And it doesn't break the tests because the error doesn't happen in the main thread. Ahh, I need a holiday soon 😅 😅